ED.cpp

```diff
@@ -234,9 +266,11 @@ Mat ED::getGradImage()
   return result8UC1;
 }
 
-int ED::getSegmentNo() { return segmentNos; }
+Mat ED::getDirImage() { return dirImage; }
```

Comment:
Return reference to reduce copy?

Reply:
I think it is good to keep it as-is as a `cv::Mat` itself behaves like a smart pointer, the operation increments its reference counter but shares the same storage.

i.e.:

```
class Hoge                                                                                                                                                                   |class Hoge
{                                                                                                                                                                            |{
public:                                                                                                                                                                      |public:
 Hoge(const std::string& path) : m_image(cv::imread(path)) {}                                                                                                                | Hoge(const std::string& path) : m_image(cv::imread(path)) {}
 ~Hoge() {}                                                                                                                                                                  | ~Hoge() {}
                                                                                                                                                                             |
 cv::Mat getImage() { return m_image; }                                                                                                                                      | cv::Mat getImage() { return m_image; }
 cv::Mat& getImageRef() { return m_image; }                                                                                                                                  | cv::Mat& getImageRef() { return m_image; }
 cv::Mat m_image;                                                                                                                                                            | cv::Mat m_image;
};                                                                                                                                                                           |};
                                                                                                                                                                             |
int main(int argc, char** argv)                                                                                                                                              |int main(int argc, char** argv)
{                                                                                                                                                                            |{
  Hoge hoge("./billiard.jpg");                                                                                                                                               |  Hoge hoge("./billiard.jpg");
  printf("%p\n", hoge.m_image.ptr());                                                                                                                                        |  printf("%p\n", hoge.m_image.ptr());
  printf("%p\n", hoge.getImage().ptr());                                                                                                                                     |  printf("%p\n", hoge.getImage().ptr());
  printf("%p\n", hoge.getImageRef().ptr());                                                                                                                                  |  printf("%p\n", hoge.getImageRef().ptr());
  return 0;                                                                                                                                                                  |  return 0;
}                                                                                                                                                                            |}
```

outputs:

```
0x7f6119cff040
0x7f6119cff040
0x7f6119cff040
```
which indicates all mat point to the same data address.

Reply:
OK, I understood.